docs/GraFx-Developers/connectors/build-media-connector/index.md

```diff
@@ -0,0 +1,284 @@
++1.417s Detected out folder: /home/parallels/connector1/node_modules/@chili-publish/connector-cli/out
++1.419s Debugger running on port 3300
+```
+
+Now open your webbrowser, and go to the IP of the machine (local or remote) that you are running this script on.
+
+In the below example, we're working on a virtual machine on ip [192.168.3.43:3300](http://192.168.3.43:3300/?type=MedieConnector).
+
+!!! tip
+	Don't forget to add '/?type=MediaConnector' to the url
+
+And the webbrowser should look like this
+
+![ui](connector04.png)
+
+Click the **Query** method, and **invoke** the query. You will get the error, that is defined to be thrown in the code.
+
+![ui](connector05.png)
+
+## Capabilities
+
+Our framework supports capabilities. You need to enable them, to expose them to the end-user.
+
+To see what is available, hover over the method, and you'll see 3 booleans that you can set.
+
+![ui](connector06.png)
+
+Let's set all 3 to **true**
+
+```typescript
+getCapabilities(): Media.MediaConnectorCapabilities {
+  return {
+	detail: true,
+	query: true,
+	filtering: true,
+}
+```
+
+## Implementation
+
+In this part, you'll build the logic that is specific for your connector.
+
+### Query
+
+E.g. for the query method, we will hardcode an example.
+
+The data structure of what needs to be returned can be found in the MediaPage definition
+
+Since the method is defined as a Promise, we'll make it async.
+
+```typescript
+async query(options: QueryOptions, context: Dictionary): Promise<Media.MediaPage> {
+
+  return {
+	pageSize: 1,
+	data:[
+	  {
+		id: '1',
+		name: 'test',
+		relativePath: 'test',
+		type: 0,
+		metaData: {},
+		extension: 'png'
+	  }
+	],
+	links: {
+	  nextPage: ''
+	}
+  }
+}
+```
+
+We'll add more logic in a moment, but for now we'll go full circle to push this to your CHILI GraFx instance.
+
+## Publish your result
+
+It's time to publish your connector to your environment.
+
+Before you can do that, you'll need to login on your instance of CHILI GraFx.
+
+Our CLI tool provides a way to authenticate via the browser. This way, your CHILI GraFx instance will recognize the cli tool, to be able to push the connector.
+
+Step 1, call the login option through the CLO
+
+```shell
+$ npx connector-cli login
++0.041s connector-cli v1.0.49
++0.042s Running command: 'login' with options: {}
++0.043s No access token found.
++0.385s Please visit https://login.chiligrafx-dev.com/activate?user_code=xxxx-yyyy and enter the code xxxx-yyyy to authenticate.
++0.385s Waiting for authentication... Ctrl+C to cancel.
+```
+
+Visit the url that is displayed, and confirm or enter the code.
+
+If succesful, the command line will show this message (with your credentials)
+
+```shell
++35.619s CLI authenticated successfully.
++35.812s User is authenticated => demo@test.com
+
+```
+
+When that is done, you can now publish.
+
+``` zsh
+npx connector-cli publish -e <environment> -b <base url> -n <connector name>
```

Comment:
@BramVerniest  in context of this PR https://github.com/chili-publish/studio-connector-contrib/pull/15, we've added possibility to specify `runtime-options` as well as proxy-options for the deployed connector. Moreover it allows you have `required` and `optional` `runtime-options` . As a result you won't be able to publish without `required` options when doing `publish` command. Might be mentioned some of them here, I think.

Reply:
@psamusev Will write down as future improvement. But would like to get v1 of this docs out